docs/multiregion.md

```diff
@@ -0,0 +1,98 @@
+# Kubecarrier multi-region proposal
+
+## Motivation
+
+Currently the KubeCarrier can only effectively operate in a single region, e.g. eu-central. Due to its cross-cluster syncing nature a high latency between management cluster and service cluster would impact the quality of service.
+
+For truly global scoped KubeCarrier installation current architecture isn't satisfactory. Here are some of the problems honest and hard-working system operator could encounter:
+
+* latency --> It's as simple as managing `eu-central` from a `east-asia` region would take between 200 and 400ms. This may or may not be satisfactory for the system requirement, but it's a matter of life. And speed of light. Certain operations require multiple round trips, increasing the overall latency.
+* Failure domains --> Failure of management cluster, or its unreachability would make the whole system immutable. However unlikely the whole management cluster becomes unavailable, it's a possible failure mode
+* Legalese & policy --> for some legal requirements/policy certain things need be separate.
+
+## Proposed solution -- introducing region concept
+
+To solve the above mentioned issues, we'll introduce managed kubecarrier federation. The basic concepts would be:
+
+* ServiceCluster --> stays the same as before; its akin to availability zone.
+* Region --> a group of multiple providers & service clusters regionally grouped with reasonable low-latency network connection is a region. The region has a single KubeCarrier management cluster
+
+This would provide the logical, geographical, and failure domain grouping to acchive the initial motivational goals. How to coordinate and configure the global KubeCarrier solution from single place (or to the end user appearing as a single place) is still an open problem.
+
+Between regions the following things should be synced, and should be globally unique:
```

Comment:
If I understood correctly, you mentioned that every region has a single KubeCarrier management cluster, and it manages a set of service clusters, why does the `ServiceCluster` should also be globally unique?

Reply:
I don't know to be honest; that's why I got from @thetechnick proposal that he wants it. 

If I had to pick it usually the region names are globally unique (look at AWS and bigger cloud providers). Also, that way we simplify our CRD names ( <service>.<service cluster>.<provider name> ) and no need adding region in the naming schema.